internal/cmd/operator-sdk/pkgmantobundle/cmd.go

```diff
@@ -0,0 +1,331 @@
+    │   ├── etcdrestore.crd.yaml
+    ├── metadata
+    │   └── annotations.yaml
+	└── tests
+        └── scorecard
+	        └── config.yaml
+
+Also, images for the both the bundles will be built with the following names: quay.io/example/etcd:0.0.1 and quay.io/example/etcd:0.0.2.
+`
+)
+
+type pkgManToBundleCmd struct {
+	// Input packagemanifest directory.
+	pkgmanifestDir string
+
+	// Optional flags for generating and building bundles.
+	outputDir string
+	baseImg   string
+	buildCmd  string
+}
+
+// NewCmd returns the pkgManToBundleCmd configured with the provided input options.
+func NewCmd() *cobra.Command {
+	p := pkgManToBundleCmd{}
+
+	pkgManToBundleCmd := &cobra.Command{
+		Use:     "pkgman-to-bundle <packagemanifestdir>",
+		Short:   "Migrates packagemanifests to bundles",
+		Long:    longHelp,
+		Example: examples,
+		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			return p.validate(args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			p.pkgmanifestDir = args[0]
+			return p.run()
+		},
+	}
+
+	pkgManToBundleCmd.Flags().StringVar(&p.outputDir, "output-dir", "", "Directory to write bundle to.")
+	pkgManToBundleCmd.Flags().StringVar(&p.baseImg, "image-base", "", "Base container image name for bundle image tags, "+
+		"ex. my.reg/foo/bar-operator will become my.reg/foo/bar-operator-bundle:${package-dir-name} for each child directory name in the packagemanifests directory")
+
+	// TODO(varsha): enable users to provide a template command so that it can be run in all child directories to build image.
+	pkgManToBundleCmd.Flags().StringVar(&p.buildCmd, "build-cmd", "", "Build command to be run for building images. By default 'docker build' is run.")
+
+	return pkgManToBundleCmd
+}
+
+// Generate the bundles from the provided packagemanifest directory.
+func (p *pkgManToBundleCmd) run() (err error) {
+	p.setDefaults()
+
+	// error if output bundle directory already exists.
+	if _, err := os.Stat(p.outputDir); !os.IsNotExist(err) {
+		return fmt.Errorf("output directory: %s for bundles already exists", p.outputDir)
+	}
+
+	// Skipping bundles here, since that's not required and could be empty for a manifest directory.
+	packages, _, err := apimanifests.GetManifestsDir(p.pkgmanifestDir)
+	if err != nil {
+		return err
+	}
+
+	if packages.IsEmpty() {
+		return fmt.Errorf("no packages found in the directory %s", p.pkgmanifestDir)
+	}
+
+	// get package metadata required for annotations.yaml and bundle.Dockerfile.
+	packageName, defaultChannel, channelsByCSV, err := getPackageMetadata(packages)
+	if err != nil {
+		return fmt.Errorf("error obtaining metadata from directory %s: %v", p.pkgmanifestDir, err)
+	}
+
+	directories, err := ioutil.ReadDir(p.pkgmanifestDir)
+	if err != nil {
+		return err
+	}
+
+	// iterate through each of the subdirectories whose name is in the valid semver format to generate bundles.
+	for _, dir := range directories {
+		if dir.IsDir() {
+			if !isValidSemver(dir.Name()) {
```

Comment:
I don't think this is a requirement for packagemanifests directories.

Reply:
Have modified this to just check for directories and convert the manifests if found